raft/src/test/java/org/apache/kafka/raft/KafkaRaftClientTest.java

```diff
@@ -137,6 +137,65 @@ public void testRejectVotesFromSameEpochAfterResigningCandidacy() throws Excepti
         context.assertSentVoteResponse(Errors.NONE, epoch, OptionalInt.empty(), false);
     }
 
+    @Test
+    public void testGrantVotesFromHigherEpochAfterResigningLeadership() throws Exception {
```

Comment:
A couple other potentially interesting test cases:

1. After starting in a resigned state, verify that the node will become a candidate after the election timer expires.
2. Verify that we can vote for new candidates when in the resigned state after shutting down.

Reply:
Thank you for your reminder, done. PTAL again.